src/Analyzers/CSharp/Analyzers/UseUTF8StringLiteral/UseUTF8StringLiteralDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,150 @@
+                new LocalizableResourceString(nameof(CSharpAnalyzersResources.Use_UTF8_string_literal), CSharpAnalyzersResources.ResourceManager, typeof(CSharpAnalyzersResources)))
+        {
+        }
+
+        protected override void InitializeWorker(AnalysisContext context)
+            => context.RegisterCompilationStartAction(context =>
+            {
+                if (context.Compilation.LanguageVersion() < LanguageVersion.Preview)
+                    return;
+
+                var expressionType = context.Compilation.GetTypeByMetadataName(typeof(System.Linq.Expressions.Expression<>).FullName!);
+
+                context.RegisterOperationAction(c => AnalyzeOperation(c, expressionType), OperationKind.ArrayCreation);
+            });
+
+        private void AnalyzeOperation(OperationAnalysisContext context, INamedTypeSymbol? expressionType)
+        {
+            var arrayCreationExpression = (IArrayCreationOperation)context.Operation;
+
+            // Don't offer if the user doesn't want it
+            var option = context.GetOption(CSharpCodeStyleOptions.PreferUTF8StringLiteral);
+            if (!option.Value)
+                return;
+
+            // Only replace arrays with initializers
+            if (arrayCreationExpression.Initializer is null)
+                return;
+
+            var byteType = context.Compilation.GetSpecialType(SpecialType.System_Byte);
+            var elementType = (arrayCreationExpression.Type as IArrayTypeSymbol)?.ElementType;
+            if (!SymbolEqualityComparer.Default.Equals(elementType, byteType))
+                return;
+
+            // UTF8 strings are not valid to use in attributes
+            if (arrayCreationExpression.Syntax.Ancestors().OfType<AttributeSyntax>().Any())
+                return;
+
+            // Can't use a UTF8 string inside an expression tree.
+            var semanticModel = context.Operation.SemanticModel;
+            Contract.ThrowIfNull(semanticModel);
+            if (arrayCreationExpression.Syntax.IsInExpressionTree(semanticModel, expressionType, context.CancellationToken))
+                return;
+
+            var values = arrayCreationExpression.Initializer.ElementValues.SelectAsArray(v => v.ConstantValue.HasValue && v.ConstantValue.Value is not null, v => v.ConstantValue.Value);
+
+            // If we couldn't get constant values for all elements then we can't offer
+            if (values.Length != arrayCreationExpression.Initializer.ElementValues.Length)
+                return;
+
+            var stringValue = GetStringValue(values);
+
+            if (stringValue is null)
+                return;
+
+            var properties = ImmutableDictionary<string, string?>.Empty.Add(StringValuePropertyName, stringValue);
+
+            context.ReportDiagnostic(
+                DiagnosticHelper.Create(Descriptor, arrayCreationExpression.Syntax.GetLocation(), option.Notification.Severity, additionalLocations: null, properties));
```

Comment:
consider putting the diagnostic on the first token, but pass along the expression syntax in additionalLocations so you can easily find the node in the fixer.

Reply:
Is it cool to just report the first token location, and not worry about the additionalLocations? `FindNode` finds the right thing anyway, since the array initializer starts in the same spot.